src/main/java/org/opensearch/agent/tools/CreateAnomalyDetectorToolEnhanced.java

```diff
@@ -0,0 +1,1402 @@
+
+            final Set<String> dateFields = ToolHelper.findDateTypeFields(fieldsToType);
+            if (dateFields.isEmpty()) {
+                listener.onFailure(new IllegalArgumentException("Index '" + indexName + "' has no date fields"));
+                return;
+            }
+
+            Map<String, String> filteredMapping = fieldsToType
+                .entrySet()
+                .stream()
+                .filter(entry -> VALID_FIELD_TYPES.contains(entry.getValue()))
+                .collect(Collectors.toUnmodifiableMap(Map.Entry::getKey, Map.Entry::getValue));
+
+            MappingContext context = new MappingContext(firstIndexName, filteredMapping, dateFields);
+            listener.onResponse(context);
+
+        }, e -> {
+            log.error("Failed to get mapping: {}", e.getMessage());
+            if (e instanceof IndexNotFoundException) {
+                listener.onFailure(new IllegalArgumentException("Index '" + indexName + "' does not exist"));
+            } else {
+                listener.onFailure(e);
+            }
+        }));
+    }
+
+    /**
+     * Step 3-6: Generate config with LLM, parse, and select date field
+     */
+    private <T> void generateAndParseConfig(
+        MappingContext mappingContext,
+        String tenantId,
+        int maxRetries,
+        ActionListener<T> listener,
+        java.util.function.BiConsumer<Map<String, String>, ActionListener<T>> nextPhaseCallback
+    ) {
+        StringJoiner dateFieldsJoiner = new StringJoiner(",");
+        mappingContext.dateFields.forEach(dateFieldsJoiner::add);
+
+        String prompt = constructPrompt(mappingContext.filteredMapping, mappingContext.indexName, dateFieldsJoiner.toString());
+
+        RemoteInferenceInputDataSet inputDataSet = RemoteInferenceInputDataSet
+            .builder()
+            .parameters(Collections.singletonMap("prompt", prompt))
+            .build();
+
+        ActionRequest request = new MLPredictionTaskRequest(
+            modelId,
+            MLInput.builder().algorithm(FunctionName.REMOTE).inputDataset(inputDataSet).build(),
+            null,
+            tenantId
+        );
+
+        client.execute(MLPredictionTaskAction.INSTANCE, request, ActionListener.wrap(mlTaskResponse -> {
+            ModelTensorOutput modelTensorOutput = (ModelTensorOutput) mlTaskResponse.getOutput();
+            ModelTensors modelTensors = modelTensorOutput.getMlModelOutputs().get(0);
+            ModelTensor modelTensor = modelTensors.getMlModelTensors().get(0);
+            Map<String, Object> dataAsMap = (Map<String, Object>) modelTensor.getDataAsMap();
+
+            if (dataAsMap == null) {
+                listener.onFailure(new IllegalStateException("Remote endpoint fails to inference."));
+                return;
+            }
+
+            String finalResponse = extractResponseFromDataAsMap(dataAsMap);
+            if (Strings.isNullOrEmpty(finalResponse)) {
```

Comment:
I like this, you should use it everwhere there's a `foo == null || foo.isEmpty()`